waku/v2/protocol/filter/waku_filter.go

```diff
@@ -189,7 +189,7 @@ func (wf *WakuFilter) FilterListener() {
 		g := new(errgroup.Group)
 		// Each subscriber is a light node that earlier on invoked
 		// a FilterRequest on this node
-		for subscriber := range wf.subscribers.Items() {
+		for subscriber := range wf.subscribers.Items(&(msg.ContentTopic)) {
 			logger := logger.With(logging.HostID("subscriber", subscriber.peer))
 			subscriber := subscriber // https://golang.org/doc/faq#closures_and_goroutines
 			if subscriber.filter.Topic != "" && subscriber.filter.Topic != topic {
```

Comment:
Note that this check will now run only for subscribers that match the content topic, instead for all subscribers. This seems like a defensive check and we're only potentially losing a warning for an unrelated subscription, which would presumably pop up in some other iteration where the content topic matches. That seems acceptable to me.